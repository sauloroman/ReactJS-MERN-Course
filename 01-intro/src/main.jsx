import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import {App} from './App.jsx'

// import './bases/01-const-let.js'
// import './bases/02-template-string.js'
// import './bases/03-objeto-literal.js'
// import './bases/04-arrays.js'
// import './bases/05-funciones.js'
// import './bases/06-deses-obj.js'
// import './bases/07-deses-arr.js'
// import './bases/08-imp-exp.js'
// import './bases/09-promesas.js'
// import './bases/10-fetch.js'
// import './bases/11-async-await.js'
import './bases/12-ternario.js'

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <App />
  </StrictMode>,
)
