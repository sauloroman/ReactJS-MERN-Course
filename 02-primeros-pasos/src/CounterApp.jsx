import { useState } from "react"
import PropTypes from "prop-types"

export const CounterApp = ({ value }) => {

  const [counter, setCounter] = useState(value)

  const onIncreaseCounter = () => {
    setCounter( counter => counter + 1 )
  }

  const onDecreaseCounter = () => {
    if ( counter === 1 ) return
    setCounter( counter =>  counter - 1 )
  }

  const onResetCounter = () => setCounter(value)

  return (
    <div>
      <h1>{counter}</h1>
      
      <button onClick={ onDecreaseCounter }>-1</button>
      <button onClick={ onResetCounter}>Reset</button>
      <button onClick={ onIncreaseCounter }>+1</button>
    </div>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}