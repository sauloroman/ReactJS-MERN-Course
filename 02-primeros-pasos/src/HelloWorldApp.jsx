export const HelloWorldApp = ({ name }) => {
  return (
    <h1 data-testid='test-title'>Hello {name}</h1>
  )
}
