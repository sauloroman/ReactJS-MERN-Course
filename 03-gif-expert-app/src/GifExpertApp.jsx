import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GifExpertApp = () => {

  const [categories, setCategories] = useState([])

  const onAddCategory = ( newCategory = '' ) => {
    if ( categories.includes( newCategory.toUpperCase() ) ) return
    setCategories( categories => [ newCategory.toUpperCase(), ...categories ] )
  }

  const onDeleteCategory = ( categoryName = '' ) => {
    setCategories( categories => categories.filter( category => category !== categoryName ))
  }

  return (
    <div className="container">
      <h1 className="title">GifExpertApp</h1>
      <AddCategory onAddCategory={onAddCategory} />

      {
        categories.map( category => (
          <GifGrid onDeleteCategory={onDeleteCategory} key={category} category={category} />
        ))
      }
    </div>
  )
}
