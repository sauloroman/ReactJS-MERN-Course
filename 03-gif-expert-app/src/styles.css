*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  box-sizing: inherit;
}

html {
  font-size: 62.5%;
}

body {
  box-sizing: border-box;
  font-size: 1.6rem;
  font-family: 'Roboto', sans-serif;
}

.btn {
  padding: .5rem;
  border-radius: 5px;
  border: none;
  background-color: #000;
  color: #fff;
  font-weight: bold;
  font-family: inherit;
  cursor: pointer;
}

.container {
  padding: 2rem 0;
  margin: 0 auto;
  width: 120rem;
  max-width: 95%;
}

.title {
  font-size: 2rem;
  letter-spacing: 2px;
  background-color: #000;
  padding: 1rem;
  color: #fff;
  margin-bottom: 2rem;
}

.form {
  width: 100%;
  margin-bottom: 2rem;
}

.form__field {
  border: 1px solid #ddd;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 2rem;
}

.form__input {
  font-family: inherit;
  width: 100%;
  padding: 0.5rem;
  border: none;
}

.form__icon {
  font-size: 2rem;
}

.category-name {
  font-size: 1.8rem;
  letter-spacing: 1px;
  margin-bottom: 2rem;
}

.gifs:not(:last-child) {
  margin-bottom: 2rem;
}

.gifs__header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 2rem;
}

.grid-gifs {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  gap: 1rem;
}

.grid-gif {
  width: 100%;
  height: 20rem;
  object-fit: cover;
}