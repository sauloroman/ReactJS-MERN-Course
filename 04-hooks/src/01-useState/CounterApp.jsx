import { useState } from 'react';

export const CounterApp = () => {
  const [{ counter1, counter2, counter3 }, setCounter] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
  });

  return (
    <>
      <h1 className="heading">UseState &mdash; CounterApp</h1>

      <div className="counters">
        <div className="counter">
          <span className="counter__value">{counter1}</span>
          <span className="counter__name">Contador 1</span>
          <button
            onClick={() =>
              setCounter((current) => ({
                ...current,
                counter1: counter1 + 1,
              }))
            }
            className="counter__btn"
          >
            Aumentar
          </button>
        </div>
        <div className="counter">
          <span className="counter__value">{counter2}</span>
          <span className="counter__name">Contador 2</span>
          <button
            onClick={() =>
              setCounter((current) => ({
                ...current,
                counter2: counter2 + 1,
              }))
            }
            className="counter__btn"
          >
            Aumentar
          </button>
        </div>
        <div className="counter">
          <span className="counter__value">{counter3}</span>
          <span className="counter__name">Contador 3</span>
          <button
            onClick={() =>
              setCounter((current) => ({
                ...current,
                counter3: counter3 + 1,
              }))
            }
            className="counter__btn"
          >
            Aumentar
          </button>
        </div>
      </div>
    </>
  );
};
