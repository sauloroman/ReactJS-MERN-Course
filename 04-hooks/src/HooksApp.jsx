// import { CounterApp } from "./01-useState/CounterApp"
// import { CounterAppWithCustomHook } from "./01-useState/CounterAppWithCustomHook"
// import { SimpleForm } from "./02-useEffect/SimpleForm"
// import { FormWithCustomHook } from "./02-useEffect/FormWithCustomHook"
// import { MultipleCustomHooks } from "./03-examples/MultipleCustomHooks"
// import { FocusScreen } from "./04-useRef/FocusScreen"
// import { Layout } from "./05-useLayoutEffect/Layout"
// import { MemoHook } from "./06-memos/MemoHook"
// import { Memorize } from "./06-memos/Memorize"
// import { CallbackHook } from "./06-memos/CallbackHook"
// import { Padre } from "./07-tarea-memo/Padre"
import { MultipleCustomHook } from "./03-examples-new/MultipleCustomHook"

export const HooksApp = () => {
  return (
    <div className="container">
      <MultipleCustomHook />
    </div>
  )
}
