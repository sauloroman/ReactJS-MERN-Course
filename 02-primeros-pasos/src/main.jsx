import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
// import { HelloWorldApp } from './HelloWorldApp'
// import { FirstApp } from './FirstApp'
import { CounterApp } from './CounterApp'
import './index.css'

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <CounterApp value={10} />
  </StrictMode>
)
