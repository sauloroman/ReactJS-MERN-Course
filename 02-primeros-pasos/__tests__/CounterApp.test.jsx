import { render, screen, fireEvent } from '@testing-library/react'
import { CounterApp } from '../src/CounterApp'

describe('Tests in CounterApp', () => {

  const initialValueCounter = 10

  test('It must match with snapshot', () => {
    const { container } = render(<CounterApp value={initialValueCounter} />)
    expect( container ).toMatchSnapshot()
  });

  test('It must show determined counter', () => {

    // const { getByText } = render(<CounterApp value={100} />)
    // expect( getByText('100') ).toBeTruthy()

    render(<CounterApp value={100} />)
    expect( screen.getByText(100) ).toBeTruthy()

  });

  test('It must increment +1 the counter when button is pressed', () => {
  
    render(<CounterApp value={initialValueCounter} />)  
    fireEvent.click( screen.getByText('+1') )
    expect( screen.getByText(`${initialValueCounter + 1}`)).toBeTruthy()

  });

  test('It must decrement -1 the counter when button is pressed', () => {

    render(<CounterApp value={initialValueCounter} />)
    fireEvent.click( screen.getByText('-1') )
    expect( screen.getByText(`${initialValueCounter - 1}`)).toBeTruthy()

  });

  test('It must reset the counter when button is pressed', () => {
    render(<CounterApp value={initialValueCounter} />)
    fireEvent.click( screen.getByText('Reset') )
    expect( screen.getByText( initialValueCounter ) ).toBeTruthy() 
  })

})