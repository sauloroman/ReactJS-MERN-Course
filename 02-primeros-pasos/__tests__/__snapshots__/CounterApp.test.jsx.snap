// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in CounterApp It must match with snapshot 1`] = `
<div>
  <div>
    <h1>
      10
    </h1>
    <button>
      -1
    </button>
    <button>
      Reset
    </button>
    <button>
      +1
    </button>
  </div>
</div>
`;
